module_utils/network_lsr/argument_validator.py

```diff
@@ -1796,6 +1796,17 @@ def _ipv4_enabled(connection):
             )
             return connection["ip"]["dhcp4"] or has_addrs4
 
```

Comment:
Proposal for the commit message summary: "Reject dns_search setting if no IPv4/IPv6 is disabled" - there is no need to distinguish between disabled/not configured, here, IMHO.

Reply:
Thanks for the proposal, I updated.